src/measure_theory/integral/bochner.lean

```diff
@@ -146,6 +145,78 @@ namespace measure_theory
 
 variables {α E F 𝕜 : Type*}
 
+section weighted_smul
+
+open continuous_linear_map
+
+variables [normed_group F] [normed_space ℝ F] {m : measurable_space α} {μ : measure α}
+
+/-- Given a set `s`, return the continuous linear map `λ x, (μ s).to_real • x`. The extension of
+that set function through `set_to_L1` gives the Bochner integral of L1 functions. -/
+def weighted_smul {m : measurable_space α} (μ : measure α) (s : set α) : F →L[ℝ] F :=
+(μ s).to_real • (continuous_linear_map.id ℝ F)
+
+lemma weighted_smul_apply {m : measurable_space α} (μ : measure α) (s : set α) (x : F) :
+  weighted_smul μ s x = (μ s).to_real • x :=
+by simp [weighted_smul]
+
+@[simp] lemma weighted_smul_zero_measure {m : measurable_space α} :
+  weighted_smul (0 : measure α) = (0 : set α → F →L[ℝ] F) :=
+by { ext1, simp [weighted_smul], }
+
+@[simp] lemma weighted_smul_empty {m : measurable_space α} (μ : measure α) :
+  weighted_smul μ ∅ = (0 : F →L[ℝ] F) :=
+by { ext1 x, rw [weighted_smul_apply], simp, }
+
+lemma weighted_smul_add_measure {m : measurable_space α} (μ ν : measure α) {s : set α}
+  (hμs : μ s ≠ ∞) (hνs : ν s ≠ ∞) :
+  (weighted_smul (μ + ν) s : F →L[ℝ] F) = weighted_smul μ s + weighted_smul ν s :=
+begin
+  ext1 x,
+  push_cast,
+  simp_rw [pi.add_apply, weighted_smul_apply],
+  push_cast,
+  rw [pi.add_apply, ennreal.to_real_add hμs hνs, add_smul],
+end
+
+lemma weighted_smul_congr (s t : set α) (hst : μ s = μ t) :
+  (weighted_smul μ s : F →L[ℝ] F) = weighted_smul μ t :=
+by { ext1 x, simp_rw weighted_smul_apply, congr' 2, }
+
+lemma weighted_smul_null {s : set α} (h_zero : μ s = 0) : (weighted_smul μ s : F →L[ℝ] F) = 0 :=
+by { ext1 x, rw [weighted_smul_apply, h_zero], simp, }
+
+lemma weighted_smul_union (s t : set α) (hs : measurable_set s) (ht : measurable_set t)
+  (hs_finite : μ s ≠ ∞) (ht_finite : μ t ≠ ∞) (h_inter : s ∩ t = ∅) :
+  (weighted_smul μ (s ∪ t) : F →L[ℝ] F) = weighted_smul μ s + weighted_smul μ t :=
+begin
+  ext1 x,
+  simp_rw [add_apply, weighted_smul_apply,
+    measure_union (set.disjoint_iff_inter_eq_empty.mpr h_inter) hs ht,
+    ennreal.to_real_add hs_finite ht_finite, add_smul],
+end
+
+lemma weighted_smul_smul [normed_field 𝕜] [normed_space 𝕜 F] [smul_comm_class ℝ 𝕜 F]
+  (c : 𝕜) (s : set α) (x : F) :
+  weighted_smul μ s (c • x) = c • weighted_smul μ s x :=
+by { simp_rw [weighted_smul_apply, smul_comm], }
+
+/-- TODO: move this. -/
+lemma norm_smul_id_le [semi_normed_group E] [nondiscrete_normed_field 𝕜] [semi_normed_space 𝕜 E]
+  (r : 𝕜) :
+  ∥r • (continuous_linear_map.id 𝕜 E)∥ ≤ ∥r∥ :=
+(norm_smul _ _).le.trans
+  ((mul_le_mul_of_nonneg_left norm_id_le (norm_nonneg _)).trans (mul_one _).le)
```

Comment:
Can you move this? :-)

Reply:
oops :)

Reply:
I removed that lemma, and changed the proof of the next one, in which that was used. I don't think it is an interesting result on its own.